packages/dashmate/test/e2e/tests/testnetMasternode.spec.js

```diff
@@ -0,0 +1,257 @@
+      container = new DockerCompose(dockerode, startedContainers);
+    });
+
+    after(async function () {
+      fs.unlinkSync(certificate.certificatePath);
+      fs.unlinkSync(certificate.privKeyPath);
+    });
+
+    it('Setup masternode', async () => {
+      const ip = await publicIp.v4();
+      const {privateKey: generatedPrivateKeyHex} = await generateBlsKeys();
+      certificate = await getSelfSignedCertificate(ip);
+      const args = `-i=${ip} -k=${generatedPrivateKeyHex} -s=manual -c=${certificate.certificatePath} -l=${certificate.privKeyPath}`;
+
+      await dashmate.setupTestnet('masternode', args);
+
+      await isTestnetServicesRunning(false, container);
+
+      const isConfig = await isConfigExist(testnetNetwork);
+      if (fs.existsSync(CONFIG_FILE_PATH) && isConfig) {
+        testnetConfig = await getConfig(testnetNetwork);
+        configEnvs = testnetConfig.toEnvs();
+      } else {
+        throw new Error(`'No configuration file in ${CONFIG_FILE_PATH}`);
+      }
+
+      if (configEnvs.CORE_MASTERNODE_ENABLE !== 'true'
+        || !configEnvs.CORE_MASTERNODE_OPERATOR_PRIVATE_KEY) {
+        throw new Error('This is not masternode configuration. Core masternode parameter is disabled.')
+      }
+    });
+
+    it('Start testnet nodes', async () => {
+      await dashmate.start(testnetNetwork);
+
+      await isTestnetServicesRunning(true, container);
+    });
+
+    it('Check core status before core sync process finish', async () => {
+      const coreStatus = await dashmate.checkStatus('core');
+      const coreOutput = JSON.parse(coreStatus.toString());
+
+      //update core version in config because versions don't match??
+      // const coreVersion = core.docker.image.replace(/\/|\(.*?\)|dashpay|dashd:|\-(.*)/g, '');
+      // expect(output.latestVersion).to.be.equal(coreVersion)
+      // try {
+      //   const latestVersionRes = await fetch('https://api.github.com/repos/dashpay/dash/releases/latest');
+      //   const latestVersion = (await latestVersionRes.json()).tag_name.substring(1);
+      //   expect(output['Latest version']).to.be.equal(latestVersion)
+      // } catch (e) {
+      //   throw e;
+      // }
+
+      expect(coreOutput.network).to.be.equal(STATUS.testnetNetwork);
+      expect(coreOutput.chain).to.be.equal(STATUS.testChain);
+      expect(coreOutput.dockerStatus).to.be.equal('running');
+      expect(coreOutput.syncAsset).to.be.equal(MasternodeSyncAssetEnum.MASTERNODE_SYNC_BLOCKCHAIN);
+
+      const peersData = await execute('docker exec dash_masternode_testnet-core-1 dash-cli getpeerinfo').then((peers) => {
+        return JSON.parse(peers.toString());
+      });
+      const peersNumber = Object.keys(peersData).length;
+      expect(coreOutput.peersCount).to.be.equal(peersNumber, 'Peers number are not matching!');
+
+      let peerHeader;
+      let peerBlock;
+      do {
+        for (const peer of peersData) {
+          peerBlock = peer['synced_blocks'];
```

Comment:
What this block is used for, I can't see that peerBlock or peerHeader is actually used somewhere?

Reply:
Actually I'm making sure that peers will sync correctly, nothing more.